Add r-deepkin 1.0.0 recipe

VonPoo opened the pull request:
**Package Summary**
Adds `r-deepkin`, an R package for efficient and precise estimation of in-depth genetic relatedness. It includes functions for calculating critical values, determining minimum effective markers, and conducting power analysis for genetic relatedness studies. This tool is useful for population genetics and kinship analysis.

**Recipe Details**
- **Version:** 1.0.0 (Synced with upstream git tag and DESCRIPTION).
- **Source:** GitHub release (v1.0.0).
- **Build:** `noarch: generic` pure R package.
- **Requirements:** `r-base >=4.3` and `r-readr`.

**Changes addressing review**
- **Cleanup:** Removed redundant Windows-specific configurations (`[win]`, `merge_build_host`) and fixed Jinja2 templating (`{{ version }}`).
- **Dependencies:** Relaxed `r-base` pinning to `>=` to allow broader compatibility.
- **Linter:** Retained `build.sh` and added `run_exports` to satisfy linter requirements for the current build setup.
- **Tests:** Simplified to a library load check to ensure installation success.

Commit:
Add r-deepkin 0.1.0 recipe (noarch R, GitHub source)

Commit:
Add r-deepkin 0.1.0 recipe (noarch R, GitHub source)

Commit:
r-deepkin: improve test diagnostics (libPaths, sessionInfo)

Commit:
r-deepkin: fix test failure by using Rscript -e (drop -q)

Commit:
r-deepkin: fix test failure by using Rscript -e (drop -q)

Commit:
r-deepkin: simplify tests to minimal library load

Commit:
r-deepkin: move tests to run_test.sh (use R -q)

Commit:
r-deepkin: bump build number to 1 and run run_test.sh explicitly

Commit:
r-deepkin: reset build number to 0 per linter

Commit:
r-deepkin: simplify tests to single R command to avoid repeated run_test.sh

Commit:
ci: retrigger r-deepkin build

bgruening commented on recipes/r-deepkin/meta.yaml:
```diff
@@ -0,0 +1,56 @@
+  sha256: 8354d2223dbc07fd11fd39270610c2a5c43146f4150a4a92185ff0ffd2eefe40
+
+build:
+  merge_build_host: True  # [win]
```
```suggestion
```

bgruening commented on recipes/r-deepkin/meta.yaml:
```diff
@@ -0,0 +1,56 @@
+  build:
+    - {{ posix }}zip               # [win]
+  host:
+    - r-base ==4.3.2
```
Do you really need this strict pined version?

VonPoo replied:
Thanks for the suggestion. I originally set the strict pin because 4.3.2 was the version I verified locally and I wanted to ensure stability. However, following your advice, I have relaxed the dependency to `>=4.3` in the latest update to allow for broader compatibility.

bgruening commented on recipes/r-deepkin/run_test.sh:
```diff
@@ -0,0 +1,5 @@
+set -euo pipefail
+
+# Minimal test: ensure the package loads successfully using the R interpreter.
+R -q -e "library('deepKin')"
```
this is doing the same as the test in the meta.yml I belief

bgruening reviewed (commented):
I assume this is AI generated? Please check the output a bit some things do not make sense in the scope of bioconda, there are redundant things included, you are not using the {{version}} variables that you declared etc ... :)

Commit:
r-deepkin: bump to 1.0.0, switch to GitHub tarball+sha256, clean windows items, add standard R build script, relax r-base >=4.3, unify tests

Commit:
r-deepkin: add noarch: generic and run_exports pin to satisfy lints

Commit:
r-deepkin: fix YAML by indenting rpaths under build

Commit:
r-deepkin: remove build.script to avoid conflict with build.sh

Commit:
r-deepkin: remove build.script to avoid conflict with build.sh

Commit:
Fix lint: remove build.sh to solve missing_run_exports error

Commit:
Fix lint: remove build.sh to solve missing_run_exports error

Commit:
Build number to 1 and force fresh CI trigger

Commit:
Build number to 0 and force fresh CI trigger

Commit:
Restore build.sh

Commit:
r-deepkin: add build.run_exports pin_subpackage(name, max_pin="x") to satisfy lint

Commit:
test commands

VonPoo commented on recipes/r-deepkin/meta.yaml:
```diff
@@ -0,0 +1,56 @@
+  recipe-maintainers:
+    - VonPoo
+  skip-lints:
+    - missing_run_exports
```
I have removed this in the latest revision.

VonPoo commented:
Thank you very much for your patience and valuable feedback.

I would like to sincerely apologize for the confusion caused by the initial submission. As this is my first time contributing to Bioconda, I was unfamiliar with the specific standards and workflow, so I relied on AI tools to generate the draft, which resulted in the redundant and non-standard configurations you pointed out.

I have now manually reviewed and refactored the entire recipe based on your guidance to ensure it meets Bioconda's standards. I am eager to learn from this process and hope to continue contributing to the community in the future.

**Summary of fixes:**
1.  **Bumped to v1.0.0:** I updated the upstream `DESCRIPTION` to `1.0.0` so the recipe version, git tag, and internal package version now all match perfectly.
2.  **Cleanup:** Removed all AI-generated hallucinations (like Windows-specific code, unnecessary `zip` deps) and fixed Jinja2 templating (`{{ version }}`).
3.  **Dependencies:** Relaxed `r-base` pinning to `>=4.3` as suggested.
4.  **Linter:** I retained `build.sh` and added `run_exports` to correctly satisfy the linter checks for this setup.
5.  **Tests:** Simplified to a functional library load check.

mencian committed:
Merge branch 'master' into add-r-deepkin-only

Commit:
chore: remove r-base version restriction for r-deepkin

Commit:
Merge branch 'master' into add-r-deepkin-only

VonPoo commented:
@mencian  Hi! I have fixed the r-base dependency issue in meta.yaml and all CI checks have passed. This recipe is now ready for review and merge. Thank you!
